smithy-aws-protocol-tests/model/restJson1/http-content-type.smithy

```diff
@@ -278,33 +278,106 @@ apply TestNoPayload @httpRequestTests([
     }
 ])
 
+/// This example POST operation serializes a request without a modeled HTTP body.
+///
+/// These tests are to ensure we do not attach a body or related headers
+/// (Content-Type) to a POST operation with no modeled payload.
+///
+@readonly
+@http(uri: "/no_payload", method: "POST")
+operation TestPostNoPayload {
+    input: TestNoPayloadInputOutput,
+    output: TestNoPayloadInputOutput
+}
+
+apply TestPostNoPayload @httpRequestTests([
+    {
+        id: "RestJsonHttpPostWithNoModeledBody",
+        documentation: "Serializes a POST request with no modeled body",
+        protocol: restJson1,
+        method: "POST",
+        uri: "/no_payload",
+        body: "",
+        forbidHeaders: [
+            "Content-Type"
+        ],
+        params: {}
+    }
+])
+
+apply TestPostNoPayload @httpRequestTests([
+    {
+        id: "RestJsonHttpWithPostHeaderMemberNoModeledBody",
+        documentation: "Serializes a POST request with header member but no modeled body",
+        protocol: restJson1,
+        method: "POST",
+        uri: "/no_payload",
+        body: "",
+        headers: {
+            "X-Amz-Test-Id": "t-12345"
+        },
+        forbidHeaders: [
+            "Content-Type"
+        ],
+        params: {
+            testId: "t-12345"
+        }
+    }
+])
+
 structure TestNoPayloadInputOutput {
     @httpHeader("X-Amz-Test-Id")
     testId: String,
 }
 
-/// This example operation has no input and serializes a request without an HTTP body.
+/// This example GET operation has no input and serializes a request without a modeled HTTP body.
 ///
 /// These tests are to ensure we do not attach a body or related headers
 /// (Content-Length, Content-Type) to operations that semantically
 /// cannot produce an HTTP body.
 ///
 @readonly
 @http(uri: "/no_input_no_payload", method: "GET")
-operation TestNoInputNoPayload {
+operation TestGetNoInputNoPayload {
     output: TestNoPayloadInputOutput
 }
 
-apply TestNoInputNoPayload @httpRequestTests([
+apply TestGetNoInputNoPayload @httpRequestTests([
     {
-        id: "RestJsonHttpWithNoInput",
+        id: "RestJsonHttpGetWithNoInput",
         documentation: "Serializes a GET request for an operation with no input, and therefore no modeled body",
         protocol: restJson1,
         method: "GET",
         uri: "/no_input_no_payload",
         body: "",
         forbidHeaders: [
-            "Content-Length",
+            "Content-Type",
+            "Content-Length"
+        ],
+        params: {}
+    }
+])
+
+/// This example POST operation has no input and serializes a request without a modeled HTTP body.
```

Comment:
nit: `modeled` is a bit odd here.

```suggestion
/// This example POST operation has no input and serializes a request without a HTTP body.
```

or just
```
/// This example POST operation has no input.
```
as the "These tests..." explains the rest.

(same for GET)